#[derive(Debug)] fails when used with a raw pointer to DST

... with the following error message:
```
error[E0277]: the trait bound `core::ops::Fn() -> bool + 'static: core::marker::Sized` is not satisfied
   --> /home/whitequark/Work/artiq-dev/artiq/artiq/runtime/../runtime.rs/src/sched.rs:109:16
    |
109 |     Completion(*const (Fn() -> bool + 'static)),
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/whitequark/Work/artiq-dev/artiq/artiq/runtime/../runtime.rs/src/sched.rs:107:10: 107:15 note: in this expansion of #[derive(Debug)] (defined in /home/whitequark/Work/artiq-dev/artiq/artiq/runtime/../runtime.rs/src/sched.rs)
    |
    = note: `core::ops::Fn() -> bool + 'static` does not have a constant size known at compile-time
    = note: required because of the requirements on the impl of `core::fmt::Debug` for `*const core::ops::Fn() -> bool + 'static`
    = note: required because of the requirements on the impl of `core::fmt::Debug` for `&*const core::ops::Fn() -> bool + 'static`
    = note: required for the cast to the object type `core::fmt::Debug`
```

since the Debug impl for raw pointers just prints the actual pointer, I don't see why it should require the pointee to be Sized.

--- a/src/libcore/fmt/mod.rs
+++ b/src/libcore/fmt/mod.rs
@@ -1566,11 +1566,11 @@ floating! { f64 }
 // Implementation of Display/Debug for various core types
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<T> Debug for *const T {
+impl<T: ?Sized> Debug for *const T {
     fn fmt(&self, f: &mut Formatter) -> Result { Pointer::fmt(self, f) }
 }
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<T> Debug for *mut T {
+impl<T: ?Sized> Debug for *mut T {
     fn fmt(&self, f: &mut Formatter) -> Result { Pointer::fmt(self, f) }
 }
 

--- a/src/test/run-pass/deriving-show.rs
+++ b/src/test/run-pass/deriving-show.rs
@@ -24,6 +24,9 @@ enum Enum {
     StructVariant { x: isize, y : usize }
 }
 
+#[derive(Debug)]
+struct Pointers(*const Send, *mut Sync);
+
 macro_rules! t {
     ($x:expr, $expected:expr) => {
         assert_eq!(format!("{:?}", $x), $expected.to_string())

